app/grandchallenge/archives/views.py

```diff
@@ -365,236 +363,82 @@ def get_context_data(self, **kwargs):
-        ),
-        Column(
-            title="Patient name",
-            sort_field="patient_name",
-            optional_condition=lambda o: o.patient_name,
-        ),
-        Column(
-            title="Patient birth date",
-            sort_field="patient_birth_date",
-            optional_condition=lambda o: o.patient_birth_date,
-        ),
-        Column(
-            title="Patient age",
-            sort_field="patient_age",
-            optional_condition=lambda o: o.patient_age,
-        ),
-        Column(
-            title="Patient sex",
-            sort_field="patient_sex",
-            optional_condition=lambda o: o.patient_sex,
-        ),
-        Column(
-            title="Study Instance UID",
-            sort_field="study_instance_uid",
-            optional_condition=lambda o: o.study_instance_uid,
-        ),
-        Column(
-            title="Study description",
-            sort_field="study_description",
-            optional_condition=lambda o: o.study_description,
-        ),
-        Column(
-            title="Study date",
-            sort_field="study_date",
-            optional_condition=lambda o: o.study_date,
-        ),
-        Column(
-            title="Series Instance UID",
-            sort_field="series_instance_uid",
-            optional_condition=lambda o: o.series_instance_uid,
-        ),
-        Column(
-            title="Series description",
-            sort_field="series_description",
-            optional_condition=lambda o: o.series_description,
-        ),
-        Column(title="View"),
-        Column(
-            title="Algorithm Results",
-            sort_field="pk",
-            optional_condition=lambda o: any(
-                civ.algorithms_jobs_as_input.exists()
-                for civ in o.componentinterfacevalue_set.all()
-            ),
-        ),
-        Column(title="Download"),
-    ]
-
-    @cached_property
-    def archive(self):
+    def base_object(self):
         return get_object_or_404(Archive, slug=self.kwargs["slug"])
 
-    def get_permission_object(self):
-        return self.archive
-
-    def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
-        context.update({"archive": self.archive})
-        return context
-
     def get_queryset(self):
-        qs = super().get_queryset()
+        queryset = super().get_queryset()
         return (
-            qs.filter(
-                componentinterfacevalue__archive_items__archive=self.archive
-            )
-            .prefetch_related(
-                "files",
-                "componentinterfacevalue_set__algorithms_jobs_as_input__algorithm_image__algorithm",
-            )
-            .select_related(
-                "origin__creator__user_profile",
-                "origin__creator__verification",
-            )
+            queryset.filter(archive=self.base_object)
+            .select_related("archive")
+            .distinct()
```

Comment:
Why is a `distinct` needed here? This does make the query slower.

Reply:
Maybe for the search? This is just for my understanding.

Reply:
I thought that since our query involves prefetching a many-to-many relationship, there is a chance of getting duplicate rows. A call to distinct() does make the query slower though and I cannot produce issues with the query right now without distinct, and examples I can find of this issue all involve filtering, so maybe this is not an issue with prefetching after all.